experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -184,8 +184,10 @@ export function createObjectProxy<TSchema extends ObjectNodeSchema, TTypes exten
 					return false;
 				}
 
-				// TODO: Is it safe to assume 'content' is a LazyEntity?
-				const field = getBoxedField(content as LazyEntity, key as FieldKey, fieldSchema);
+				// TODO: Is it safe to assume 'content' is a LazyObjectNode?
+				assert(content instanceof LazyObjectNode, "invalid content");
```

Comment:
When unsure an assumption the code depends on always holds, its much better to narrow with an assert rather than `as` so we will learn if its ever not true instead of having incorrectly hyped behavior at runtime which can be hard to debug.